internals/overlord/state/change.go

```diff
@@ -607,3 +783,87 @@ func (c *Change) abortTasks(tasks []*Task, abortedLanes map[int]bool, seenTasks
 		c.abortLanes(lanes, abortedLanes, seenTasks)
 	}
 }
+
+type TaskDependencyCycleError struct {
+	IDs []string
+	msg string
+}
+
+func (e *TaskDependencyCycleError) Error() string { return e.msg }
+
+func (e *TaskDependencyCycleError) Is(err error) bool {
```

Comment:
Weird that this is defined -- isn't this the default for `errors.Is`?

Reply:
Wow. I could not find any obvious reason for this in the history. The original author has gone already.

That seems to be defined to make [some tests work](https://github.com/snapcore/snapd/blob/5d02ea78acef7bfb8bd2ce50c7874c895acde832/overlord/state/change_test.go#L1114) (which, probably, should have been done with the explicit type assertion).

Should we keep it, raise with snapd and pick the commit from snapd/overlord/state pkg?

Reply:
No, let's keep it as is, in line with snapd. Could raise with snapd, but it's very minor -- I probably wouldn't bother.

Reply:
I think this pattern happened in snapd without me noticing it propely/thinking too much about it, I think the general direction of the errors package was that:

`if err == ErrFoo ...`  generalizes to `if errors.Is(err, ErrFoo)`
`if _, ok := err.(*FooError); ok`  generalizes to `var fooErr *FooError; if errors.As(err, fooErr) ...`

I think in snapd people started introducing this kind of `Is` implementation to avoid using the more complex `As` and just do instead  `if errors.Is(err, &FooError{})`.  I'm not sure how much the latter is a bit outside of the original intentions `errors`  vs bad style.